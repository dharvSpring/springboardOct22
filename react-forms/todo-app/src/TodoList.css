.TodoList {
    margin-left: 1rem;
    font-size: large;
}