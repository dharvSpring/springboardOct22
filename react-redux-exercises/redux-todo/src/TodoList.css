.TodoList {
    margin: 3rem;
}

input, button {
    padding: 0.3rem;
}