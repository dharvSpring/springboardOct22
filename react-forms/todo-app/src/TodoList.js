import React, { useState } from "react";
import './TodoList.css';

import Todo from "./Todo";
import NewTodoForm from './NewTodoForm';

function TodoList() {
    const [todoList, setTodoList] = useState([]);

    const addTodo = ({id, todoText}) => {
        setTodoList([
            ...todoList,
            {id, text: todoText},
        ]);
    }

    const removeTodo = (id) => {
        setTodoList(todoList.filter(todo => (todo.id !== id)));
    }

    const todoListComp = todoList.map(todo => {
        return (
            <Todo key={todo.id} id={todo.id} text={todo.text} handleRemove={removeTodo} />
        )
    })

    return (
        <div className="TodoList">
            <h1>Todo!</h1>
            <NewTodoForm handleSubmit={addTodo}/>
            {todoListComp}
        </div>
    )
}

export default TodoList;