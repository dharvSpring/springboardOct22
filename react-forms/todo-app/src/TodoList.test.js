import React from "react";
import { render, fireEvent } from "@testing-library/react";

import TodoList from "./TodoList";

function addTodo(todoList, text="submit to github") {
    const todoComp = todoList.getByLabelText("New Todo");
    fireEvent.change(todoComp, { target: { value: text }});

    const submitComp = todoList.getByText("Add Todo!");
    fireEvent.click(submitComp);
}

it("smoketest", function() {
    render(<TodoList />);
});

it("snapshot", function() {
    const { asFragment } = render(<TodoList />);
    expect(asFragment()).toMatchSnapshot();
});

it("add todo", function() {
    const list = render(<TodoList />);
    addTodo(list);
    
    expect(list.getByText("submit to github")).toBeInTheDocument();
    expect(list.getByText("X")).toBeInTheDocument();
    
    // form should be empty
    expect(list.getByLabelText("New Todo")).toHaveValue("");
});

it("delete todo", function() {
    const list = render(<TodoList />);
    addTodo(list);
    
    fireEvent.click(list.getByText("X"));
    expect(list.queryByText("submit to github")).not.toBeInTheDocument();
});