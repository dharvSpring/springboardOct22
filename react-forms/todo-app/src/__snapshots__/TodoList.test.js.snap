// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<DocumentFragment>
  <div
    class="TodoList"
  >
    <h1>
      Todo!
    </h1>
    <form
      class="NewTodoForm"
    >
      <label
        for="todoText"
      >
        New Todo
      </label>
      <input
        id="todoText"
        name="todoText"
        type="text"
        value=""
      />
      <input
        type="submit"
        value="Add Todo!"
      />
    </form>
  </div>
</DocumentFragment>
`;
